File: app/src/main/java/com/flaringapp/base/presentation/activities/main/impl/MainPresenter.java
package com.flaringapp.base.presentation.activities.main.impl;

import androidx.annotation.Nullable;

import com.flaringapp.base.presentation.activities.main.IMainPresenter;
import com.flaringapp.base.presentation.activities.main.IMainView;
import com.flaringapp.base.presentation.activities.main.navigation.Navigator;
import com.flaringapp.base.presentation.activities.main.navigation.Screen;
import com.flaringapp.base.presentation.mvp.BasePresenter;

public class MainPresenter extends BasePresenter<IMainView> implements IMainPresenter, Navigator {

    @Override
    public void navigateTo(Screen screen, @Nullable Object... data) {
        if (view != null) {
            view.openScreen(screen, data);
        }
    }
}

File: app/src/main/java/com/flaringapp/base/presentation/activities/impl/MainPresenter.kt

package com.flaringapp.base.presentation.activities.impl

import com.flaringapp.base.presentation.activities.MainContract
import com.flaringapp.base.presentation.activities.navigation.Screen
import com.flaringapp.base.presentation.fragments.home.impl.HomeFragment
import com.flaringapp.base.presentation.mvp.BasePresenter

class MainPresenter: BasePresenter<MainContract.ViewContract>(), MainContract.PresenterContract {

    override fun onNavigate(screen: Screen, data: Any?) {
        val fragment = when(screen) {
            Screen.HOME -> HomeFragment.newInstance()
        }
        view?.openScreen(fragment)
    }
}